interfaces/ITodo.ts

```diff
@@ -11,7 +11,7 @@ export interface ITodo {
    status: EStatuses;
    description: string;
    dateAdded: string;
-   dateCompleted: string;
+   dateCompleted: string | null;
```

Comment:
Q-time: where does null come into play and why? if you don't need this property all the time then you can make it an optional one like we do with id. You could also be getting to the point where you can get a little fancier with your interfaces and make a baseline interface that gets implemented by or extended by other interfaces for specific uses.

The difference would be in the "strictness" of your type safety, in general we want to be explicit and as a strict as possible other wise we undermine the point of typescript. So if a property doesn't have a valid use case for being two things then it shouldn't be two things. So date completed can exist as a string or not exist vs data completed can be a string or null, which one do you want and why